EL8 ppc64le FAIL: TestCgroups

### Version of Singularity:

What version of Singularity are you using? Run:

```
3.7.x
```

### Actual behavior

```
=== RUN   TestCgroups
        cgroups_linux_test.go:64: open /sys/fs/cgroup/hugetlb/singularity/154890/hugetlb.16MB.limit_in_bytes: permission denied
    --- FAIL: TestCgroups (0.42s)
```

This 16MB hugetlb was specifically set for RHEL7 ppc64le in https://github.com/hpcng/singularity/pull/5918

However, on EL 8.3 we see the otherwise standard 2MB...

```
[centos@centos8-ppc64le ~]$ ls /sys/fs/cgroup/hugetlb/singularity/154890/
cgroup.clone_children           hugetlb.2MB.failcnt
cgroup.procs                    hugetlb.2MB.limit_in_bytes
hugetlb.1GB.failcnt             hugetlb.2MB.max_usage_in_bytes
hugetlb.1GB.limit_in_bytes      hugetlb.2MB.usage_in_bytes
hugetlb.1GB.max_usage_in_bytes  notify_on_release
hugetlb.1GB.usage_in_bytes      tasks
```

--- a/internal/pkg/cgroups/cgroups_linux_test.go
+++ b/internal/pkg/cgroups/cgroups_linux_test.go
@@ -12,7 +12,6 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
-	"runtime"
 	"strconv"
 	"strings"
 	"testing"
@@ -55,9 +54,12 @@ func TestCgroups(t *testing.T) {
 	manager := &Manager{Pid: pid, Path: path}
 
 	cgroupsToml := "example/cgroups.toml"
-	// A different pageSize is needed on pp64le
-	if runtime.GOARCH == "ppc64le" {
-		cgroupsToml = "example/cgroups-ppc64le.toml"
+	// Some systems, e.g. ppc64le may not have a 2MB page size, so don't
+	// apply a 2MB hugetlb limit if that's the case.
+	_, err = os.Stat("/sys/fs/cgroup/hugetlb/hugetlb.2MB.limit_in_bytes")
+	if os.IsNotExist(err) {
+		t.Log("No hugetlb.2MB.limit_in_bytes - using alternate cgroups test file")
+		cgroupsToml = "example/cgroups-no-hugetlb.toml"
 	}
 
 	if err := manager.ApplyFromFile(cgroupsToml); err != nil {

--- a/internal/pkg/cgroups/example/cgroups-no-hugetlb.toml
+++ b/internal/pkg/cgroups/example/cgroups-no-hugetlb.toml
@@ -102,9 +102,9 @@
 # Hugetlb limit (in bytes)
 # - pagesize: the hugepage size
 # - limit: the limit of "hugepagesize" hugetlb usage
-[[hugepageLimits]]
-  limit = 9223372036854771712
-  pageSize = "16MB"
+# [[hugepageLimits]]
+#   limit = 9223372036854771712
+#   pageSize = "16MB"
 
 
 # Network restriction configuration

